kratos/utilities/normal_calculation_utils.cpp

```diff
@@ -201,6 +201,43 @@ void NormalCalculationUtils::CalculateOnSimplex(
 /***********************************************************************************/
 /***********************************************************************************/
 
+void NormalCalculationUtils::CalculateShapeDerivativesOnSimplexConditions(
+    ConditionsArrayType& rConditions,
+    const std::size_t Dimension
+    )
+{
+    KRATOS_TRY
+
+    if (Dimension == 2) {
+        // reset NORMAL_SHAPE_DERIVATIVE variable for all conditions
+        VariableUtils().SetNonHistoricalVariable(NORMAL_SHAPE_DERIVATIVE,
+                                                 ZeroMatrix(4, 2), rConditions);
+
+        // calculate condition normal shape derivatives
+        BlockPartition<ConditionsArrayType>(rConditions).for_each([](ConditionType& rCondition) {
+            if (rCondition.GetGeometry().PointsNumber() == 2) {
+                CalculateNormalShapeDerivative2D(rCondition);
+            }
+        });
+    } else {
+        // reset NORMAL_SHAPE_DERIVATIVE variable for all conditions
+        VariableUtils().SetNonHistoricalVariable(NORMAL_SHAPE_DERIVATIVE,
+                                                 ZeroMatrix(9, 3), rConditions);
+
+        // calculate condition normal shape derivatives
+        BlockPartition<ConditionsArrayType>(rConditions).for_each([](ConditionType& rCondition) {
+            if (rCondition.GetGeometry().PointsNumber() == 3) {
```

Comment:
does this also work for quadratic lines? => I recommend checking against the geom type

Reply:
done :)